packages/frontend/src/page/index.tsx

```diff
@@ -32,21 +36,45 @@ export const Component = ({
 }: KPITokenRemotePageProps): ReactElement => {
     const publicClient = usePublicClient();
     const { data: walletClient } = useWalletClient();
-    const { chain } = useNetwork();
+    const { chain } = useAccount();
 
-    const { data: tokenData, isLoading: loadingTokenData } = useToken({
-        address: kpiToken?.address as Address,
-        staleTime: 2_000,
+    const { data: tokenData, isLoading: loadingTokenData } = useReadContracts({
+        contracts: [
+            {
+                address: kpiToken?.address,
+                abi: erc20Abi,
+                functionName: "name",
+            },
+            {
+                address: kpiToken?.address,
+                abi: erc20Abi,
+                functionName: "decimals",
+            },
+            {
+                address: kpiToken?.address,
+                abi: erc20Abi,
+                functionName: "symbol",
+            },
+            {
```

Comment:
I'm not sure why we're fetching `totalSupply` here when there's a hook later that fetches it with watch mode enabled.